cosmic_ray/importing.py

```diff
@@ -61,8 +61,11 @@ def using_mutant(module_name, mutant):
     mutated AST.
 
     """
-    if module_name in sys.modules:
-        del sys.modules[module_name]
+    module_to_delete = ""
+    for parent in module_name.split('.'):
+        module_to_delete = ("%s.%s" % (module_to_delete, parent)).lstrip('.')
+        if module_to_delete in sys.modules:
+            del sys.modules[module_to_delete]
```

Comment:
Interesting. Something must be holding a reference to the original.

On Wed, Jul 20, 2016, 12:31 Alexander Todorov <notifications@github.com>
wrote:

> In cosmic_ray/importing.py
> <https://github.com/sixty-north/cosmic-ray/pull/158#discussion_r71501432>:
>
> > @@ -61,8 +61,11 @@ def using_mutant(module_name, mutant):
> >      mutated AST.
> >
> >      """
> > -    if module_name in sys.modules:
> > -        del sys.modules[module_name]
> > +    module_to_delete = ""
> > +    for parent in module_name.split('.'):
> > +        module_to_delete = ("%s.%s" % (module_to_delete, parent)).lstrip('.')
> > +        if module_to_delete in sys.modules:
> > +            del sys.modules[module_to_delete]
>
> For some reason the assignment to sys.modules doesn't have the desired
> effect. Looping through del_mods and deleting them one by one however seems
> to be fine.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/sixty-north/cosmic-ray/pull/158/files/c7ab6a4cad1cda12495a235d173c8c3f2cf2b610#r71501432>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAE1eC9cOAD8eEsEdNYlnCksmlE8Vj2Kks5qXfjngaJpZM4JQjEd>
> .
>